Platform/CLinuxOperatingSystemStats/CLinuxOperatingSystemStats.c

```diff
@@ -10,6 +10,51 @@
 
 #include <stdio.h>
 #include "CLinuxOperatingSystemStats.h"
+#include <linux/perf_event.h>    /* Definition of PERF_* constants */
+#include <linux/hw_breakpoint.h> /* Definition of HW_* constants */
+#include <sys/syscall.h>         /* Definition of SYS_* constants */
+#include <unistd.h>
+#include <string.h> // memset
+#include <sys/ioctl.h>
+#include <errno.h>
+
+int CLinuxPerformanceCountersInit() {
+    int fd;
+    struct perf_event_attr  pe;
+
+    memset(&pe, 0, sizeof(pe));
+    pe.type = PERF_TYPE_HARDWARE;
+    pe.size = sizeof(pe);
+    pe.config = PERF_COUNT_HW_INSTRUCTIONS;
+    pe.disabled = 1;
+    pe.exclude_kernel = 1;
+    pe.exclude_hv = 1;
+
+    fd = syscall(SYS_perf_event_open, &pe, 0, -1, -1, 0);
+    if (fd == -1) {
+        perror("Error in perf_event_open syscall");
+        fprintf(stderr, "Error opening leader %llx\n", pe.config);
+        fprintf(stderr, "Detailed errno: %s\n", strerror(errno));
```

Comment:
I'd save `errno` right after failed syscall, so it's not overwritten by subsequent `perror()/fprintf()` calls.